client/my-sites/patterns/components/header/index.tsx

```diff
@@ -1,23 +1,180 @@
+import { Substitution, translate } from 'i18n-calypso';
+import DocumentHead from 'calypso/components/data/document-head';
 import { preventWidows } from 'calypso/lib/formatting';
 import { PatternsSearchField } from 'calypso/my-sites/patterns/components/search-field';
+import type { Category } from 'calypso/my-sites/patterns/types';
 
 import './style.scss';
 
-type PatternsHeaderProps = {
-	description: string;
-	title: string;
+const OPTIONS = {
+	home: {
+		title: {
+			comment: 'HTML title of the page that showcases block patterns',
+			textOnly: true,
+		},
+		header: {
+			comment: 'Header of the page that showcases block patterns',
+			textOnly: true,
+		},
+		description: {
+			comment: 'Description of the page that showcases block patterns',
+			textOnly: true,
+		},
+	},
+	category: {
+		title: {
+			comment: 'HTML title of the page that lists block patterns for a specific category',
+			textOnly: true,
+		},
+		header: {
+			comment: 'Header of the page that lists block patterns for a specific category',
+			textOnly: true,
+		},
+		description: {
+			comment: 'Description of the page that lists block patterns for a specific category',
+			textOnly: true,
+		},
+	},
 };
 
-export const PatternsHeader = ( { description, title }: PatternsHeaderProps ) => {
+const CONTENT: Record<
+	Category[ 'name' ],
+	{ description: Substitution; header: Substitution; title: Substitution }
```

Comment:
I had to switch the type from `string` to `Substitution` to please TypeScript but maybe there is a better approach.